File: Shared/Components/BadgeUnlockOverlay.swift
//
//  Untitled.swift
//  WatchVerse
//
//  Created by Maxime on 2026-08-12.
//

import SwiftUI

struct BadgeUnlockOverlay: View {

    let badge: Badge
    let onClose: () -> Void
    let onSeeBadge: () -> Void

    var body: some View {

        ZStack {

            Color.black.opacity(0.35)
                .ignoresSafeArea()

            VStack(spacing: 24) {

                Image(badge.imageName)
                    .resizable()
                    .scaledToFit()
                    .frame(width: 160, height: 160)

                Text("Badge Unlocked!")
                    .font(.title.bold())
                    .foregroundStyle(.white)

                Text(badge.title)
                    .font(.title2.bold())
                    .foregroundStyle(Color.watchVerseGold)

                Text(badge.description)
                    .foregroundStyle(.gray)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal)

                HStack(spacing: 16) {

                    Button {
                        onClose()
                    } label: {
                        Text("Close")
                            .frame(maxWidth: .infinity)
                            .frame(height: 50)
                            .background(.gray.opacity(0.3))
                            .clipShape(RoundedRectangle(cornerRadius: 14))
                    }

                    Button {
                        onSeeBadge()
                    } label: {
                        Text("See Badge")
                            .frame(maxWidth: .infinity)
                            .frame(height: 50)
                            .background(Color.watchVerseGold)
                            .foregroundStyle(.black)
                            .clipShape(RoundedRectangle(cornerRadius: 14))
                    }
                }
                .padding(.horizontal)

            }
            .padding()
            .background {
                RoundedRectangle(cornerRadius: 24)
                    .fill(.ultraThinMaterial)
                    .overlay {
                        RoundedRectangle(cornerRadius: 24)
                            .stroke(Color.white.opacity(0.15), lineWidth: 1)
                    }
            }
            .padding()
        }
    }
}

File: Android/WatchVerse/app/src/main/java/com/maximeproulx/watchverse/BadgeUnlockOverlay.kt
package com.maximeproulx.watchverse

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

private val BadgeUnlockGold = Color(
    red = 0.87f,
    green = 0.74f,
    blue = 0.28f
)

@Composable
fun BadgeUnlockOverlay(
    badge: Badge,
    onClose: () -> Unit,
    onSeeBadge: () -> Unit
) {
    val context = LocalContext.current
    val imageResource = context.resources.getIdentifier(
        badge.imageName,
        "drawable",
        context.packageName
    )
    val backdropInteraction = remember {
        MutableInteractionSource()
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.Black.copy(alpha = 0.35f))
            .clickable(
                interactionSource = backdropInteraction,
                indication = null,
                onClick = {}
            ),
        contentAlignment = Alignment.Center
    ) {
        Column(
            modifier = Modifier
                .padding(16.dp)
                .fillMaxWidth()
                .clip(RoundedCornerShape(24.dp))
                .background(Color(0xEE242424))
                .border(
                    width = 1.dp,
                    color = Color.White.copy(alpha = 0.15f),
                    shape = RoundedCornerShape(24.dp)
                )
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(24.dp)
        ) {
            Image(
                painter = painterResource(imageResource),
                contentDescription = badge.title,
                modifier = Modifier.size(160.dp),
                contentScale = ContentScale.Fit
            )

            Text(
                text = "Badge Unlocked!",
                color = Color.White,
                fontSize = 28.sp,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center
            )

            Text(
                text = badge.title,
                color = BadgeUnlockGold,
                fontSize = 22.sp,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center
            )

            Text(
                text = badge.description,
                color = Color.Gray,
                fontSize = 16.sp,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(horizontal = 16.dp)
            )

            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                horizontalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                Box(
                    modifier = Modifier
                        .weight(1f)
                        .height(50.dp)
                        .clip(RoundedCornerShape(14.dp))
                        .background(Color.Gray.copy(alpha = 0.30f))
                        .clickable(onClick = onClose),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "Close",
                        color = Color.White,
                        fontWeight = FontWeight.Medium
                    )
                }

                Box(
                    modifier = Modifier
                        .weight(1f)
                        .height(50.dp)
                        .clip(RoundedCornerShape(14.dp))
                        .background(BadgeUnlockGold)
                        .clickable(onClick = onSeeBadge),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "See Badge",
                        color = Color.Black,
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        }
    }
}
